modules/codegen/src/smithy4s/codegen/Renderer.scala

```diff
@@ -581,23 +612,27 @@ private[codegen] class Renderer(compilationUnit: CompilationUnit) { self =>
             val cn = caseName(a)
             // format: off
             lines(
+              deprecationAnnotation(altHints),
               line"case object $cn extends $name",
               line"""private val ${cn}Alt = $Schema_.constant($cn)${renderConstraintValidation(altHints)}.oneOf[$name]("$realName").addHints(hints)""",
               line"private val ${cn}AltWithValue = ${cn}Alt($cn)"
             )
             // format: on
-          case a @ Alt(altName, _, UnionMember.TypeCase(tpe), _) =>
+          case a @ Alt(altName, _, UnionMember.TypeCase(tpe), altHints) =>
             val cn = caseName(a)
             lines(
+              deprecationAnnotation(altHints),
               line"case class $cn(${uncapitalise(altName)}: $tpe) extends $name"
             )
-          case Alt(_, realName, UnionMember.ProductCase(struct), _) =>
+          case Alt(_, realName, UnionMember.ProductCase(struct), altHints) =>
             val additionalLines = lines(
               newline,
               line"""val alt = schema.oneOf[$name]("$realName")"""
             )
             renderProduct(
-              struct,
+              // putting alt hints first should result in higher priority of these.
+              // might need deduplication (although the Hints type will take care of it, just in case)
+              struct.copy(hints = altHints ++ struct.hints),
```

Comment:
(I'll add this as a comment later)

I'm putting the deprecation note on the struct because there's nothing else to put it on.